File: src/main/js/shuffle.js
const shuffle = require('fisher-yates');
const ParkMiller = require('park-miller');
const fs = require('fs');
const Stream = require('stream');

const lemario = fs.readFileSync('../resources/data/lemario.txt', 'utf8').split("\n");

const range = max => {
  const numbers = [];
  while (numbers.length < max)
    numbers.push(numbers.length + 1);
  return numbers;
};

const randomNumber = digits => () => Math.trunc(Math.random() * 10 ** digits);

const seeds = range(100).map(randomNumber(10));
fs.writeFileSync('../resources/data/seeds.txt', seeds.join("\n"));
seeds.forEach(seed => {
  const readable = Stream.Readable();
  const writeable = fs.createWriteStream(`../resources/shuffle/js/lemario_shuffle_${seed}.txt`, {flags: 'w'});
  readable.pipe(writeable);

  const random = new ParkMiller(seed);
  // Got to use old function because shuffle
  // depends on scope switching
  function nextFloat() {
    return random.float();
  }

  shuffle(lemario, nextFloat).forEach(line => readable.push(`${line}\n`));
  readable.push(null);
});

File: src/main/java/Shuffle.java
import static java.nio.file.StandardOpenOption.CREATE;
import static java.nio.file.StandardOpenOption.TRUNCATE_EXISTING;
import static java.util.stream.Collectors.joining;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.List;

public class Shuffle {
  public static void main(String[] args) throws IOException {
    Path shuffleDir = Paths.get("src/main/resources/shuffle/java");
    List<String> lemario = Files.readAllLines(Paths.get("src/main/resources/data/lemario.txt"));
    Files.lines(Paths.get("src/main/resources/data/seeds.txt"))
        .map(Long::parseLong)
        .forEach(seed -> {
          uncheckedWrite(
              shuffleDir.resolve("lemario_shuffle_" + seed + ".txt"),
              FisherYates.shuffle(lemario, new ParkMiller(seed)).stream().collect(joining("\n"))
          );
        });
  }

  private static void uncheckedWrite(Path path, String content) {
    try {
      Files.write(path, content.getBytes(), CREATE, TRUNCATE_EXISTING);
    } catch (IOException e) {
      throw new RuntimeException(e);
    }
  }
}
